tests/Makefile

```diff
@@ -8,6 +8,17 @@ endif
 # Attempt to guess where to find various tools and libraries
 ################################################################
 
+MODE ?= release
+ifeq ($(MODE),release)
+	UC_MODE := Release
+else ifeq ($(MODE),debug)
+	UC_MODE := Debug
+else ifeq ($(MODE),fast)
+	UC_MODE := Fast
+else
+	$(error "Mode must be 'debug' or 'release'")
+endif
```

Comment:
The xcode projects don't seem to like 'fast' mode anyway... Also, I don't think the bin_dir in the top-level makefile is quite correct. We shouldn't have to copy binaries at all during the build - things should use them from where they are built to. Perhaps this is something we should sort out in a separate PR - something to add to the backlog?